integrations/hanlp/src/haystack_integrations/components/preprocessors/hanlp/chinese_document_splitter.py

```diff
@@ -0,0 +1,499 @@
+            than the threshold, it's attached to the previous split.
+        :param respect_sentence_boundary: Choose whether to respect sentence boundaries when splitting by "word".
+            If True, uses HanLP to detect sentence boundaries, ensuring splits occur only between sentences.
+        :param splitting_function: Necessary when `split_by` is set to "function".
+            This is a function which must accept a single `str` as input and return a `list` of `str` as output,
+            representing the chunks after splitting.
+        :param granularity: The granularity of Chinese word segmentation, either 'coarse' or 'fine'.
+
+        :raises ValueError: If the granularity is not 'coarse' or 'fine'.
+        """
+        self.split_by = split_by
+        self.split_length = split_length
+        self.split_overlap = split_overlap
+        self.split_threshold = split_threshold
+        self.respect_sentence_boundary = respect_sentence_boundary
+        self.splitting_function = splitting_function
+        self.granularity = granularity
+
+        if granularity not in {"coarse", "fine"}:
+            msg = f"Invalid granularity '{granularity}'. Choose either 'coarse' or 'fine'."
+            raise ValueError(msg)
+
+    def run(self, documents: List[Document]) -> Dict[str, List[Document]]:
+        """
+        Split documents into smaller chunks.
+
+        :param documents: The documents to split.
+        :return: A dictionary containing the split documents.
+
+        :raises RuntimeError: If the Chinese word segmentation model is not loaded.
+        """
+        if self.split_sent is None:
+            msg = "The Chinese word segmentation model is not loaded. Please run 'warm_up()' before calling 'run()'."
+            raise RuntimeError(msg)
+
+        split_docs = []
+        for doc in documents:
+            split_docs.extend(self._split_document(doc))
+        return {"documents": split_docs}
+
+    def warm_up(self) -> None:
+        """Warm up the component by loading the necessary models."""
+        if self.granularity == "coarse":
+            self.chinese_tokenizer = hanlp.load(hanlp.pretrained.tok.COARSE_ELECTRA_SMALL_ZH)
+        if self.granularity == "fine":
+            self.chinese_tokenizer = hanlp.load(hanlp.pretrained.tok.FINE_ELECTRA_SMALL_ZH)
+        self.split_sent = hanlp.load(hanlp.pretrained.eos.UD_CTB_EOS_MUL)
+
+    def _split_by_character(self, doc: Document) -> List[Document]:
+        """
+        Define a function to handle Chinese clauses
+
+        :param doc: The document to split.
+        :return: A list of split documents.
+        """
+        if doc.content is None:
+            return []
+
+        split_at = _CHARACTER_SPLIT_BY_MAPPING[self.split_by]
+
+        units = self.chinese_tokenizer(doc.content)
+
+        for i in range(len(units) - 1):  # pylint: disable=possibly-used-before-assignment
```

Comment:
`# pylint: disable=possibly-used-before-assignment` 

there's no pylint running in CI - I suspect this can be safely removed, probably leftover from being in the haystack main